src/index.ts

```diff
@@ -74,18 +111,25 @@ const Vessel = (
     };
 
   const getModal = () => {
-    let modal = document.getElementById(GLOBAL_MODAL_ID) as HTMLIFrameElement;
-    if (!modal) {
-      modal = addModal();
-      // Must be attached exactly once or we'll send multiple messages
-      window.addEventListener('message', initHandler(modal));
-    }
-
+    const modal =
+      (document.getElementById(GLOBAL_MODAL_ID) as HTMLIFrameElement) ??
+      addModal();
     return modal;
   };
 
   const modal: HTMLIFrameElement = getModal();
 
+  if (!window.__vessel_sdk) {
+    throw new VesselError('Invalid state: the vessel sdk was destroyed');
+  }
+
+  if (window.__vessel_sdk?.callback) {
```

Comment:
curious, why the remove and then the add instead of just adding if it doesn't exist?

Reply:
This is the fix for the issue they reported where the callbacks weren't updated if the `Vessel(...)` func was called many times with different different callback functions passed.

Now, every time the function is called we update the DOM and set the new callback functions.